VictoriaAlbert/VictoriaAlbert/Museum.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  Museum.swift
+//  VictoriaAlbert
+//
+//  Created by Harichandan Singh on 11/10/16.
+//  Copyright © 2016 Harichandan Singh. All rights reserved.
+//
+
+import Foundation
+
+enum ParsingErrors: Error {
+    case recordsArrayError, fieldsDictError, objectError, dateTextError, placeError, titleError, primaryImageIDError, artistError, locationError, museumNumberError, yearError
+}
+class Museum {
+    //MARK: - Properties
+    let object: String
+    let dateText: String
+    let place: String
+    let title: String
+    let primaryImageID: String
+    let artist: String
+    let location: String
+    let museumNumber: String
+    let year: Int
+    
+    //MARK: - Initializers
+    init(object: String, dateText: String, place: String, title: String, primaryImageID: String, artist: String, location: String, museumNumber: String, year: Int) {
+        self.object = object
+        self.dateText = dateText
+        self.place = place
+        self.title = title
+        self.primaryImageID = primaryImageID
+        self.artist = artist
+        self.location = location
+        self.museumNumber = museumNumber
+        self.year = year
+    }
+    
+    //MARK: - Methods
+    static func turnDataIntoMuseumArray(data: Data) -> [Museum]? {
+        do {
+            //get JSON Data
+            let jsonData = try JSONSerialization.jsonObject(with: data, options: [])
+            
+            //Parsing data
+            guard let museumDict = jsonData as? [String: Any] else {
+                print("There was an error casting the JSON data!")
+                return nil
+            }
+            print("Data successfully casted into museumDict.")
+            
+            guard let recordsArray = museumDict["records"] as? [[String: Any]] else {
+                throw ParsingErrors.recordsArrayError
+            }
+            print("Data successfully casted into recordsArray.")
+            
+            var allMuseums: [Museum] = []
+            
+            //loop through array to find property values
+            for dict in recordsArray {
+                guard let fieldsDict = dict["fields"] as? [String: Any] else {
+                    throw ParsingErrors.fieldsDictError
+                }
+                
+                guard let object = fieldsDict["object"] as? String else {
+                    throw ParsingErrors.objectError
+                }
+                
+                guard let dateText = fieldsDict["date_text"] as? String else {
+                    throw ParsingErrors.dateTextError
+                }
+                
+                guard let place = fieldsDict["place"] as? String else {
```

Comment:
Are you sure you want to guard let and throw for all fields? I think some fields could remain optional. Also you could  ```continue``` in the else block so that any one object that has a missing field wouldn't prevent the whole array from forming.